### ISSUE TITLE
SSACSolver does not compile on win10

### ISSUE THREAD

**Issue by georgemhall** [GillesPy2/GillesPy2#21]
BuildError: Error encountered while compiling file:
Return code: 2.
Error:
b'UserSimulation.cpp: In member function \xe2\x80\x98virtual double PropensityFunction::evaluate(uint, uint*)\xe2\x80\x99:\nUserSimulation.cpp:36:28: error: expected \xe2\x80\x98;\xe2\x80\x99 before \xe2\x80\x98a\xe2\x80\x99\n             return state[0]a*vol;\n                            ^\nUserSimulation.cpp:36:28: error: \xe2\x80\x98a\xe2\x80\x99 was not declared in this scope\nUserSimulation.cpp:39:20: error: \xe2\x80\x98Mustate\xe2\x80\x99 was not declared in this scope\n             return Mustate[1]*state[1];\n                    ^\nmake: *** [UserSimulation] Error 1\n'

@briandrawert @Thisisnotdalton

**Comment by Thisisnotdalton** [comment]
This seems to be a substring-name issue. The NumPy and C++ SSA solvers were made with the assumption that the parameter and species names would not be substrings of one another. We will look into solving this issue in future changes, but for now, you should be able to avoid this issue by making sure that your species and parameter names are not substrings of one another.

**Comment by Thisisnotdalton** [comment]
We'll work on writing a model.clean_propensities function which will handle taking care of any name substring conflicts for when solvers simulate the model.

**Comment by briandrawert** [comment]
@georgemhall can you check again with the most recent develop branch?  @seanebum can you replicate?  If fixed, please close.

**Comment by briandrawert** [comment]
@georgemhall please run on Win10, and close or document errors.

**Comment by Thisisnotdalton** [comment]
I have tested running the latest develop version on Windows using the Windows Subsystem for Linux and the SSACSolver ran the example notebooks without issue.

**Comment by briandrawert** [comment]
@georgemhall please run "/examples/Michaelis-Menten/Michaelis-Menten_SSA_C.ipynb" if it runs, close this issue, if not post errors.

**Comment by georgemhall** [comment]
When running jupyter notebook from the root (GillesPy2) folder, I get the following error when trying to run the Michaelis-Menten_SSA_C notebook:

`---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-2-f7040cbfb905> in <module>()
----> 1 import gillespy2
      2 class MichaelisMenten(gillespy2.Model):
      3      def __init__(self, parameter_values=None):
      4             #initialize Model
      5             gillespy2.Model.__init__(self, name="Michaelis_Menten")

ModuleNotFoundError: No module named 'gillespy2'`

Which of course was fixed by adding a new cell under the initial imports to append the path:
`import sys
import os
sys.path.append(os.path.abspath(os.path.join(os.getcwd(), '../../')))
print(sys.path)`

When it hits the cell adding the SSACsolver/model, I get this:

`---------------------------------------------------------------------------
BuildError                                Traceback (most recent call last)
<ipython-input-6-0501060c5ee6> in <module>()
      1 from gillespy2.solvers.cpp import SSACSolver
      2 model = MichaelisMenten()
----> 3 csolver = SSACSolver(model)

j:\Code\GP2_issue-21\GillesPy2\gillespy2\solvers\cpp\ssa_c_solver.py in __init__(self, model, output_directory, delete_directory)
    126             copy_files(self.output_directory)
    127             self.write_template()
--> 128             self.compile()
    129 
    130     def __del__(self):

j:\Code\GP2_issue-21\GillesPy2\gillespy2\solvers\cpp\ssa_c_solver.py in compile(self)
    158             self.compiled = True
    159         else:
--> 160             raise gillespyError.BuildError("Error encountered while compiling file:\nReturn code: {0}.\nError:\n{1}\n".format(built.returncode, built.stderr))
    161 
    162     def run(self=None, model=None, t=20, number_of_trajectories=1,

BuildError: Error encountered while compiling file:
Return code: 2.
Error:
b'In file included from model.cpp:1:0:\r\nmodel.h:12:5: error: \'uint\' does not name a type\r\n     uint id; //useful for index id in arrays\r\n     ^~~~\r\nmodel.h:14:5: error: \'uint\' does not name a type\r\n     uint initial_population;\r\n     ^~~~\r\nmodel.h:18:5: error: \'uint\' does not name a type\r\n     uint id; //useful for propensity function id associated\r\n     ^~~~\r\nmodel.h:21:19: error: \'uint\' was not declared in this scope\r\n     std :: vector<uint> affected_reactions; //list of which reactions have propensities that would change with this reaction firing\r\n                   ^~~~\r\nmodel.h:21:23: error: template argument 1 is invalid\r\n     std :: vector<uint> affected_reactions; //list of which reactions have propensities that would change with this reaction firing\r\n                       ^\r\nmodel.h:21:23: error: template argument 2 is invalid\r\nIn file included from model.cpp:1:0:\r\nmodel.h:26:5: error: \'uint\' does not name a type\r\n     uint number_species;\r\n     ^~~~\r\nmodel.h:28:5: error: \'uint\' does not name a type\r\n     uint number_reactions;\r\n     ^~~~\r\nmodel.h:30:69: error: \'uint\' was not declared in this scope\r\n     Model(std :: vector<std :: string> species_names, std :: vector<uint> species_populations, std :: vector<std :: string> reaction_names);\r\n                                                                     ^~~~\r\nmodel.h:30:73: error: template argument 1 is invalid\r\n     Model(std :: vector<std :: string> species_names, std :: vector<uint> species_populations, std :: vector<std :: string> reaction_names);\r\n                                                                         ^\r\nmodel.h:30:73: error: template argument 2 is invalid\r\nmodel.h:37:29: error: \'uint\' has not been declared\r\n     virtual double evaluate(uint reaction_number, uint* state) = 0;\r\n                             ^~~~\r\nmodel.h:37:51: error: \'uint\' has not been declared\r\n     virtual double evaluate(uint reaction_number, uint* state) = 0;\r\n                                                   ^~~~\r\nmodel.h:48:5: error: \'uint\' does not name a type\r\n     uint number_timesteps;\r\n     ^~~~\r\nmodel.h:49:5: error: \'uint\' does not name a type\r\n     uint number_trajectories;\r\n     ^~~~\r\nmodel.h:50:5: error: \'uint\' does not name a type\r\n     uint* trajectories_1D;\r\n     ^~~~\r\nmodel.h:51:5: error: \'uint\' does not name a type\r\n     uint*** trajectories;\r\n     ^~~~\r\nmodel.h:53:30: error: \'uint\' has not been declared\r\n     Simulation(Model* model, uint number_trajectories, uint number_timesteps, double end_time, IPropensityFunction* propensity_function, int random_seed);\r\n                              ^~~~\r\nmodel.h:53:56: error: \'uint\' has not been declared\r\n     Simulation(Model* model, uint number_trajectories, uint number_timesteps, double end_time, IPropensityFunction* propensity_function, int random_seed);\r\n                                                        ^~~~\r\nmodel.cpp:5:76: error: \'uint\' was not declared in this scope\r\n   Model :: Model(std :: vector<std :: string> species_names, std :: vector<uint> species_populations, std :: vector<std :: string> reaction_names):\r\n                                                                            ^~~~\r\nmodel.cpp:5:80: error: template argument 1 is invalid\r\n   Model :: Model(std :: vector<std :: string> species_names, std :: vector<uint> species_populations, std :: vector<std :: string> reaction_names):\r\n                                                                                ^\r\nmodel.cpp:5:80: error: template argument 2 is invalid\r\nmodel.cpp: In constructor \'Gillespy::Model::Model(std::vector<std::__cxx11::basic_string<char> >, int, std::vector<std::__cxx11::basic_string<char> >)\':\r\nmodel.cpp:6:5: error: class \'Gillespy::Model\' does not have any field named \'number_species\'\r\n     number_species(species_names.size()),\r\n     ^~~~~~~~~~~~~~\r\nmodel.cpp:7:5: error: class \'Gillespy::Model\' does not have any field named \'number_reactions\'\r\n     number_reactions(reaction_names.size())\r\n     ^~~~~~~~~~~~~~~~\r\nmodel.cpp:9:45: error: \'number_species\' was not declared in this scope\r\n     species = std :: make_unique<Species[]>(number_species);\r\n                                             ^~~~~~~~~~~~~~\r\nmodel.cpp:10:9: error: \'uint\' was not declared in this scope\r\n     for(uint i = 0; i < number_species; i++){\r\n         ^~~~\r\nmodel.cpp:10:21: error: \'i\' was not declared in this scope\r\n     for(uint i = 0; i < number_species; i++){\r\n                     ^\r\nmodel.cpp:15:48: error: \'number_reactions\' was not declared in this scope\r\n     reactions = std :: make_unique<Reaction[]>(number_reactions);\r\n                                                ^~~~~~~~~~~~~~~~\r\nmodel.cpp:16:9: error: \'uint\' was not declared in this scope\r\n     for(uint i = 0; i < number_reactions; i++){\r\n         ^~~~\r\nmodel.cpp:16:21: error: \'i\' was not declared in this scope\r\n     for(uint i = 0; i < number_reactions; i++){\r\n                     ^\r\nmodel.cpp:19:16: error: expected \';\' before \'j\'\r\n       for(uint j = 0; j < number_species; j++){\r\n                ^\r\nmodel.cpp:19:23: error: \'j\' was not declared in this scope\r\n       for(uint j = 0; j < number_species; j++){\r\n                       ^\r\nmodel.cpp:22:55: error: the value of \'uint\' is not usable in a constant expression\r\n       reactions[i].affected_reactions = std :: vector<uint>();\r\n                                                       ^~~~\r\nmodel.cpp:16:9: note: \'uint\' was not declared \'constexpr\'\r\n     for(uint i = 0; i < number_reactions; i++){\r\n         ^~~~\r\nmodel.cpp:22:59: error: type/value mismatch at argument 1 in template parameter list for \'template<class _Tp, class _Alloc> class std::vector\'\r\n       reactions[i].affected_reactions = std :: vector<uint>();\r\n                                                           ^\r\nmodel.cpp:22:59: note:   expected a type, got \'uint\'\r\nmodel.cpp:22:59: error: template argument 2 is invalid\r\nmodel.cpp: In member function \'void Gillespy::Model::update_affected_reactions()\':\r\nmodel.cpp:28:9: error: \'uint\' was not declared in this scope\r\n     for(uint i = 0; i < number_reactions; i++){\r\n         ^~~~\r\nmodel.cpp:28:21: error: \'i\' was not declared in this scope\r\n     for(uint i = 0; i < number_reactions; i++){\r\n                     ^\r\nmodel.cpp:28:25: error: \'number_reactions\' was not declared in this scope\r\n     for(uint i = 0; i < number_reactions; i++){\r\n                         ^~~~~~~~~~~~~~~~\r\nmodel.cpp:32:9: error: \'uint\' was not declared in this scope\r\n     for(uint r1 = 0; r1 < number_reactions; r1++){\r\n         ^~~~\r\nmodel.cpp:32:22: error: \'r1\' was not declared in this scope\r\n     for(uint r1 = 0; r1 < number_reactions; r1++){\r\n                      ^~\r\nmodel.cpp:32:27: error: \'number_reactions\' was not declared in this scope\r\n     for(uint r1 = 0; r1 < number_reactions; r1++){\r\n                           ^~~~~~~~~~~~~~~~\r\nmodel.cpp:34:16: error: expected \';\' before \'r2\'\r\n       for(uint r2 = r1 + 1; r2 < number_reactions; r2++){\r\n                ^~\r\nmodel.cpp:34:29: error: \'r2\' was not declared in this scope\r\n       for(uint r2 = r1 + 1; r2 < number_reactions; r2++){\r\n                             ^~\r\nmodel.cpp:35:11: error: expected \';\' before \'s\'\r\n  for(uint s = 0; s < number_species; s++){\r\n           ^\r\nmodel.cpp:35:18: error: \'s\' was not declared in this scope\r\n  for(uint s = 0; s < number_species; s++){\r\n                  ^\r\nmodel.cpp:35:22: error: \'number_species\' was not declared in this scope\r\n  for(uint s = 0; s < number_species; s++){\r\n                      ^~~~~~~~~~~~~~\r\nmodel.cpp: At global scope:\r\nmodel.cpp:46:42: error: \'uint\' has not been declared\r\n   Simulation :: Simulation(Model* model, uint number_trajectories, uint number_timesteps, double end_time, IPropensityFunction* propensity_function, int random_seed) : model(model), end_time(end_time), random_seed(random_seed), number_timesteps(number_timesteps), number_trajectories(number_trajectories), propensity_function(propensity_function){\r\n                                          ^~~~\r\nmodel.cpp:46:68: error: \'uint\' has not been declared\r\n   Simulation :: Simulation(Model* model, uint number_trajectories, uint number_timesteps, double end_time, IPropensityFunction* propensity_function, int random_seed) : model(model), end_time(end_time), random_seed(random_seed), number_timesteps(number_timesteps), number_trajectories(number_trajectories), propensity_function(propensity_function){\r\n                                                                    ^~~~\r\nmodel.cpp: In constructor \'Gillespy::Simulation::Simulation(Gillespy::Model*, int, int, double, Gillespy::IPropensityFunction*, int)\':\r\nmodel.cpp:46:229: error: class \'Gillespy::Simulation\' does not have any field named \'number_timesteps\'\r\n   Simulation :: Simulation(Model* model, uint number_trajectories, uint number_timesteps, double end_time, IPropensityFunction* propensity_function, int random_seed) : model(model), end_time(end_time), random_seed(random_seed), number_timesteps(number_timesteps), number_trajectories(number_trajectories), propensity_function(propensity_function){\r\n                                                                                                                                                                                                                                     ^~~~~~~~~~~~~~~~\r\nmodel.cpp:46:265: error: class \'Gillespy::Simulation\' does not have any field named \'number_trajectories\'\r\n   Simulation :: Simulation(Model* model, uint number_trajectories, uint number_timesteps, double end_time, IPropensityFunction* propensity_function, int random_seed) : model(model), end_time(end_time), random_seed(random_seed), number_timesteps(number_timesteps), number_trajectories(number_trajectories), propensity_function(propensity_function){\r\n                                                                                                                                                                                                                                                                         ^~~~~~~~~~~~~~~~~~~\r\nmodel.cpp:49:9: error: \'uint\' was not declared in this scope\r\n     for(uint i = 0; i < number_timesteps; i++){\r\n         ^~~~\r\nmodel.cpp:49:21: error: \'i\' was not declared in this scope\r\n     for(uint i = 0; i < number_timesteps; i++){\r\n                     ^\r\nmodel.cpp:52:5: error: \'uint\' was not declared in this scope\r\n     uint trajectory_size = number_timesteps * (model -> number_species);\r\n     ^~~~\r\nmodel.cpp:53:5: error: \'trajectories_1D\' was not declared in this scope\r\n     trajectories_1D = new uint[number_trajectories * trajectory_size];\r\n     ^~~~~~~~~~~~~~~\r\nmodel.cpp:53:27: error: \'uint\' does not name a type\r\n     trajectories_1D = new uint[number_trajectories * trajectory_size];\r\n                           ^~~~\r\nmodel.cpp:54:5: error: \'trajectories\' was not declared in this scope\r\n     trajectories = new uint**[number_trajectories];\r\n     ^~~~~~~~~~~~\r\nmodel.cpp:54:24: error: \'uint\' does not name a type\r\n     trajectories = new uint**[number_trajectories];\r\n                        ^~~~\r\nmodel.cpp: In lambda function:\r\nmodel.cpp:54:51: error: expected \'{\' before \';\' token\r\n     trajectories = new uint**[number_trajectories];\r\n                                                   ^\r\nmodel.cpp: In constructor \'Gillespy::Simulation::Simulation(Gillespy::Model*, int, int, double, Gillespy::IPropensityFunction*, int)\':\r\nmodel.cpp:54:29: error: no match for \'operator*\' (operand type is \'Gillespy::Simulation::Simulation(Gillespy::Model*, int, int, double, Gillespy::IPropensityFunction*, int)::<lambda()>\')\r\n     trajectories = new uint**[number_trajectories];\r\n                              \r\nmodel.cpp:55:14: error: expected \';\' before \'i\'\r\n     for(uint i = 0; i < number_trajectories; i++){\r\n              ^\r\nmodel.cpp:55:21: error: \'i\' was not declared in this scope\r\n     for(uint i = 0; i < number_trajectories; i++){\r\n                     ^\r\nmodel.cpp:56:29: error: \'uint\' does not name a type\r\n       trajectories[i] = new uint*[number_timesteps];\r\n                             ^~~~\r\nmodel.cpp: In lambda function:\r\nmodel.cpp:56:52: error: expected \'{\' before \';\' token\r\n       trajectories[i] = new uint*[number_timesteps];\r\n                                                    ^\r\nmodel.cpp: In constructor \'Gillespy::Simulation::Simulation(Gillespy::Model*, int, int, double, Gillespy::IPropensityFunction*, int)\':\r\nmodel.cpp:57:16: error: expected \';\' before \'j\'\r\n       for(uint j = 0; j < number_timesteps; j++){\r\n                ^\r\nmodel.cpp:57:23: error: \'j\' was not declared in this scope\r\n       for(uint j = 0; j < number_timesteps; j++){\r\n                       ^\r\nmodel.cpp:58:45: error: \'trajectory_size\' was not declared in this scope\r\n  trajectories[i][j] = &(trajectories_1D[i * trajectory_size + j *  (model -> number_species)]);\r\n                                             ^~~~~~~~~~~~~~~\r\nmodel.cpp:58:78: error: \'struct Gillespy::Model\' has no member named \'number_species\'; did you mean \'species\'?\r\n  trajectories[i][j] = &(trajectories_1D[i * trajectory_size + j *  (model -> number_species)]);\r\n                                                                              ^~~~~~~~~~~~~~\r\nmodel.cpp: In destructor \'Gillespy::Simulation::~Simulation()\':\r\nmodel.cpp:66:12: error: \'trajectories_1D\' was not declared in this scope\r\n     delete trajectories_1D;\r\n            ^~~~~~~~~~~~~~~\r\nmodel.cpp:67:9: error: \'uint\' was not declared in this scope\r\n     for(uint i = 0; i < number_trajectories; i++){\r\n         ^~~~\r\nmodel.cpp:67:21: error: \'i\' was not declared in this scope\r\n     for(uint i = 0; i < number_trajectories; i++){\r\n                     ^\r\nmodel.cpp:67:25: error: \'number_trajectories\' was not declared in this scope\r\n     for(uint i = 0; i < number_trajectories; i++){\r\n                         ^~~~~~~~~~~~~~~~~~~\r\nmodel.cpp:68:14: error: \'trajectories\' was not declared in this scope\r\n       delete trajectories[i];\r\n              ^~~~~~~~~~~~\r\nmodel.cpp:70:12: error: \'trajectories\' was not declared in this scope\r\n     delete trajectories;\r\n            ^~~~~~~~~~~~\r\nmodel.cpp: In function \'std::ostream& Gillespy::operator<<(std::ostream&, const Gillespy::Simulation&)\':\r\nmodel.cpp:75:9: error: \'uint\' was not declared in this scope\r\n     for(uint i = 0; i < simulation.number_timesteps; i++){\r\n         ^~~~\r\nmodel.cpp:75:21: error: \'i\' was not declared in this scope\r\n     for(uint i = 0; i < simulation.number_timesteps; i++){\r\n                     ^\r\nmodel.cpp:75:36: error: \'const struct Gillespy::Simulation\' has no member named \'number_timesteps\'\r\n     for(uint i = 0; i < simulation.number_timesteps; i++){\r\n                                    ^~~~~~~~~~~~~~~~\r\nmodel.cpp:77:16: error: expected \';\' before \'trajectory\'\r\n       for(uint trajectory = 0; trajectory < simulation.number_trajectories; trajectory++){\r\n                ^~~~~~~~~~\r\nmodel.cpp:77:32: error: \'trajectory\' was not declared in this scope\r\n       for(uint trajectory = 0; trajectory < simulation.number_trajectories; trajectory++){\r\n                                ^~~~~~~~~~\r\nmodel.cpp:77:56: error: \'const struct Gillespy::Simulation\' has no member named \'number_trajectories\'\r\n       for(uint trajectory = 0; trajectory < simulation.number_trajectories; trajectory++){\r\n                                                        ^~~~~~~~~~~~~~~~~~~\r\nmodel.cpp:78:11: error: expected \';\' before \'j\'\r\n  for(uint j = 0; j < simulation.model -> number_species; j++){\r\n           ^\r\nmodel.cpp:78:18: error: \'j\' was not declared in this scope\r\n  for(uint j = 0; j < simulation.model -> number_species; j++){\r\n                  ^\r\nmodel.cpp:78:42: error: \'struct Gillespy::Model\' has no member named \'number_species\'; did you mean \'species\'?\r\n  for(uint j = 0; j < simulation.model -> number_species; j++){\r\n                                          ^~~~~~~~~~~~~~\r\nmodel.cpp:79:21: error: \'const struct Gillespy::Simulation\' has no member named \'trajectories\'\r\n    os << simulation.trajectories[trajectory][i][j] <<  " ";\r\n                     ^~~~~~~~~~~~\r\nmodel.cpp: In member function \'void Gillespy::Simulation::output_results_buffer(std::ostream&)\':\r\nmodel.cpp:90:9: error: \'uint\' was not declared in this scope\r\n     for(uint i = 0; i < number_timesteps; i++){\r\n         ^~~~\r\nmodel.cpp:90:21: error: \'i\' was not declared in this scope\r\n     for(uint i = 0; i < number_timesteps; i++){\r\n                     ^\r\nmodel.cpp:90:25: error: \'number_timesteps\' was not declared in this scope\r\n     for(uint i = 0; i < number_timesteps; i++){\r\n                         ^~~~~~~~~~~~~~~~\r\nmodel.cpp:92:16: error: expected \';\' before \'byte_i\'\r\n       for(uint byte_i = 0; byte_i < sizeof(double); byte_i++){\r\n                ^~~~~~\r\nmodel.cpp:92:28: error: \'byte_i\' was not declared in this scope\r\n       for(uint byte_i = 0; byte_i < sizeof(double); byte_i++){\r\n                            ^~~~~~\r\nmodel.cpp:95:16: error: expected \';\' before \'trajectory\'\r\n       for(uint trajectory = 0; trajectory < number_trajectories; trajectory++){\r\n                ^~~~~~~~~~\r\nmodel.cpp:95:32: error: \'trajectory\' was not declared in this scope\r\n       for(uint trajectory = 0; trajectory < number_trajectories; trajectory++){\r\n                                ^~~~~~~~~~\r\nmodel.cpp:95:45: error: \'number_trajectories\' was not declared in this scope\r\n       for(uint trajectory = 0; trajectory < number_trajectories; trajectory++){\r\n                                             ^~~~~~~~~~~~~~~~~~~\r\nmodel.cpp:96:11: error: expected \';\' before \'j\'\r\n  for(uint j = 0; j < model -> number_species; j++){\r\n           ^\r\nmodel.cpp:96:18: error: \'j\' was not declared in this scope\r\n  for(uint j = 0; j < model -> number_species; j++){\r\n                  ^\r\nmodel.cpp:96:31: error: \'struct Gillespy::Model\' has no member named \'number_species\'; did you mean \'species\'?\r\n  for(uint j = 0; j < model -> number_species; j++){\r\n                               ^~~~~~~~~~~~~~\r\nmodel.cpp:97:11: error: \'trajectories\' was not declared in this scope\r\n    temp = trajectories[trajectory][i][j];\r\n           ^~~~~~~~~~~~\r\nmodel.cpp:98:13: error: expected \';\' before \'byte_i\'\r\n    for(uint byte_i = 0; byte_i < sizeof(double); byte_i++){\r\n             ^~~~~~\r\nmodel.cpp:98:25: error: \'byte_i\' was not declared in this scope\r\n    for(uint byte_i = 0; byte_i < sizeof(double); byte_i++){\r\n                         ^~~~~~\r\nmake: *** [model.o] Error 1\n'`

**Comment by Thisisnotdalton** [comment]
Can you provide me with information about the compiler you're using? Additionally, can you try to instantiate the SSACSolver with a specified output directory and specifying delete_directory=False so we can see the generated C++ source?

**Comment by georgemhall** [comment]
Output of gcc -v:

```
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=J:/Program\ Files/mingw-w64/x86_64-6.3.0-posix-seh-rt_v5-rev1/mingw64/bin/../libexec/gcc/x86_64-w64-mingw32/6.3.0/lto-wrapper.exe
Target: x86_64-w64-mingw32
Configured with: ../../../src/gcc-6.3.0/configure --host=x86_64-w64-mingw32 --build=x86_64-w64-mingw32 --target=x86_64-w64-mingw32 --prefix=/mingw64 --with-sysroot=/c/mingw630/x86_64-630-posix-seh-rt_v5-rev1/mingw64 --enable-shared --enable-static --disable-multilib --enable-languages=c,c++,fortran,lto --enable-libstdcxx-debug --enable-libstdcxx-time=yes --enable-threads=posix --enable-libgomp --enable-libatomic --enable-lto --enable-graphite --enable-checking=release --enable-fully-dynamic-string --enable-version-specific-runtime-libs --enable-libstdcxx-filesystem-ts=yes --disable-libstdcxx-pch --disable-libstdcxx-debug --enable-bootstrap --disable-rpath --disable-win32-registry --disable-nls --disable-werror --disable-symvers --with-gnu-as --with-gnu-ld --with-arch=nocona --with-tune=core2 --with-libiconv --with-system-zlib --with-gmp=/c/mingw630/prerequisites/x86_64-w64-mingw32-static --with-mpfr=/c/mingw630/prerequisites/x86_64-w64-mingw32-static --with-mpc=/c/mingw630/prerequisites/x86_64-w64-mingw32-static --with-isl=/c/mingw630/prerequisites/x86_64-w64-mingw32-static --with-pkgversion='x86_64-posix-seh-rev1, Built by MinGW-W64 project' --with-bugurl=http://sourceforge.net/projects/mingw-w64 CFLAGS='-O2 -pipe -fno-ident -I/c/mingw630/x86_64-630-posix-seh-rt_v5-rev1/mingw64/opt/include -I/c/mingw630/prerequisites/x86_64-zlib-static/include -I/c/mingw630/prerequisites/x86_64-w64-mingw32-static/include' CXXFLAGS='-O2 -pipe -fno-ident -I/c/mingw630/x86_64-630-posix-seh-rt_v5-rev1/mingw64/opt/include -I/c/mingw630/prerequisites/x86_64-zlib-static/include -I/c/mingw630/prerequisites/x86_64-w64-mingw32-static/include' CPPFLAGS=' -I/c/mingw630/x86_64-630-posix-seh-rt_v5-rev1/mingw64/opt/include -I/c/mingw630/prerequisites/x86_64-zlib-static/include -I/c/mingw630/prerequisites/x86_64-w64-mingw32-static/include' LDFLAGS='-pipe -fno-ident -L/c/mingw630/x86_64-630-posix-seh-rt_v5-rev1/mingw64/opt/lib -L/c/mingw630/prerequisites/x86_64-zlib-static/lib -L/c/mingw630/prerequisites/x86_64-w64-mingw32-static/lib '
Thread model: posix
gcc version 6.3.0 (x86_64-posix-seh-rev1, Built by MinGW-W64 project)
```

Output files:
[c_solver_output.zip](https://github.com/GillesPy2/GillesPy2/files/2771424/c_solver_output.zip)

**Comment by georgemhall** [comment]
I'm supposed to get a graph from matplotlib right? This is what I get:

In:
```
model = MichaelisMenten()
results = model.run(show_labels=False)
dresults = model.run(solver=csolver, show_labels=False)
plt.plot(dresults[0][:,0],dresults[0][:,1],'--r', label='Substrate')
plt.plot(dresults[0][:,0],dresults[0][:,4],'--b', label='Product')
plt.plot(results[0][:,0],results[0][:,1],'-r', label='Substrate')
plt.plot(results[0][:,0],results[0][:,4],'-b', label='Product')
plt.plot([0],[11])
plt.legend(loc='best')
```
Out:
`<matplotlib.legend.Legend at 0x27043143080>`

**Comment by georgemhall** [comment]
Compiling seems to work now. The issue I'm running into now is that there's no matplotlib graph showing for the first plt.plot cell. I double checked, and matplotlib, numpy and seaborn are all installed for python 3:
```
>py --version
Python 3.7.0

>py -m pip list
Package         Version
--------------- -------
coverage        4.5.1
cycler          0.10.0
Cython          0.29.1
kiwisolver      1.0.1
matplotlib      2.2.3
numpy           1.15.1
pandas          0.23.4
pip             10.0.1
pyparsing       2.2.0
python-dateutil 2.7.3
pytz            2018.5
scipy           1.1.0
seaborn         0.9.0
setuptools      39.0.1
six             1.11.0
virtualenv      16.0.0
```

**Comment by georgemhall** [comment]
@briandrawert once the above PR #53 is merged we can close this issue

**Comment by briandrawert** [comment]
Closed with #53
